DAQ/src/ArtBinaryPacketsFromDigis_module.cc

```diff
@@ -0,0 +1,1113 @@
+    int panel = SD.strawId().getPanel();
+    int plane = SD.strawId().getPlane();
+
+    // ROC ID, counting from 0 across all DTCs (for the tracker)
+    //    uint8_t localROCID = panel;
+    uint8_t globalROCID = (plane * 6) + panel;
+
+    // 240 ROCs total
+    if (globalROCID >= number_of_rocs) {
+      throw cet::exception("DATA") << " Global ROC ID " << globalROCID
+				   << " exceeds limit of " << number_of_rocs;
+    }
+    HeaderData.ROCID = panel;
+    HeaderData.unused1 = 0;
+    HeaderData.SubsystemID = DTCLib::DTC_Subsystem_Tracker;
+    HeaderData.Valid = 1;
+    // Word 2
+    HeaderData.PacketCount = 2;
+    HeaderData.unused2 = 0;
+    // Word 3
+    uint64_t timestamp = EventNum;
+    HeaderData.TimestampLow = static_cast<adc_t>(timestamp & 0xFFFF);
+    // Word 4
+    HeaderData.TimestampMed = static_cast<adc_t>((timestamp >> 16) & 0xFFFF);
+    // Word 5
+    HeaderData.TimestampHigh = static_cast<adc_t>((timestamp >> 32) & 0xFFFF);
+    // Word 6
+    HeaderData.Status = 0; // 0 corresponds to "TimeStamp had valid data"
+    HeaderData.FormatVersion = format_version;
+    // Word 7
+    HeaderData.DTCID = static_cast<uint8_t>(globalROCID / number_of_rocs_per_dtc);
+    uint8_t  evbMode = 0;//ask Eric
+    HeaderData.EVBMode = evbMode;
+
+  }
+
+  void   ArtBinaryPacketsFromDigis::fillTrackerDataPacket(const StrawDigi& SD, TrackerDataPacket& TrkData) {
+
+    TrkData.StrawIndex = SD.strawId().asUint16();
+    TrkData.TDC0 = SD.TDC(StrawEnd::cal);
+    TrkData.TDC1 = SD.TDC(StrawEnd::hv);
+    TrkData.TOT0 = SD.TOT(StrawEnd::cal);
+    TrkData.TOT1 = SD.TOT(StrawEnd::hv);
+
+    TrkTypes::ADCWaveform const& theWaveform = SD.adcWaveform();
+
+    TrkData.SetWaveform(0,  theWaveform[0]);
+    TrkData.SetWaveform(1,  theWaveform[1]);
+    TrkData.SetWaveform(2,  theWaveform[2]);
+    TrkData.SetWaveform(3,  theWaveform[3]);
+    TrkData.SetWaveform(4,  theWaveform[4]);
+    TrkData.SetWaveform(5,  theWaveform[5]);
+    TrkData.SetWaveform(6,  theWaveform[6]);
+    TrkData.SetWaveform(7,  theWaveform[7]);  
+    TrkData.SetWaveform(8,  theWaveform[8]);  
+    TrkData.SetWaveform(9,  theWaveform[9]);  
+    TrkData.SetWaveform(10, theWaveform[10]);  
+    TrkData.SetWaveform(11, theWaveform[11]); 
+    TrkData.SetWaveform(12, theWaveform[12]); 
+    TrkData.SetWaveform(13, theWaveform[13]); 
+    TrkData.SetWaveform(14, theWaveform[14]); 
+
+    TrkData.PreprocessingFlags = 0;
+
+    //    TrkData.unused1 = ;
+    if (_enableFPGAEmulation) {
+      // Note: Eventually, there will be a calibration database to provide these sorts of values
+      // For now, these are just placeholders
+      calib_constant_type clockstart = 320; //10 / .03125 (tdclsb)
+      calib_constant_type maxenergyls8 = 583, minenergyls8 = 0;
+      calib_constant_type gainrs15 = 1389;
+      calib_constant_type inverseionizationenergyls26 = 633;
+      calib_constant_type panelTDCoffset = 0, hvoffset = 0, caloffset = 0;
+
```

Comment:
Magic numbers again.

Reply:
that block of code is something from Tomo. I kept in only for keeping track, but so far I have never used that nor I read anything in Tomo's documentation about it. 
I can either delete it or comment it.
It seems to reproduce the operations performed already on the DRAC. I have never heard about any use-case for doing that.

Reply:
If you expect that you will want this functionality soon, then keep it.  If you do not expect to need that functionality then please remove it but put enough information in your commit comment that you can find it again if you do need it.